### PR TITLE
Add relative path support for visualization file names.

### PR DESCRIPTION
**Pull request by MTCam**
This adds relative file path support for visualization file names.

Without this change, specifying relative filenames for the visualization file works OK, except for the filenames listed in the parallel manifest, which had previous retained the relative filenames.  This change massages the filenames listed in the parallel manifest so as to drop the relative pathing.

### CODE REVIEW

File: meshmode/discretization/visualization.py [hunk 1]
```diff
@@ -639,7 +641,7 @@ def write_parallel_vtk_file(self, mpi_comm, file_name_pattern, names_and_fields,
                 use_high_order=use_high_order,
                 par_manifest_filename=par_manifest_filename,
                 par_file_names=[
-                    file_name_pattern.format(rank=rank)
+                    os.path.basename(file_name_pattern.format(rank=rank))
```
Comment (inducer) [inline meshmode/discretization/visualization.py:644]: AFAICT, this breaks the logic that currently determines which process is responsible for writing the manifest:
https://github.com/inducer/meshmode/blob/9058582b695091f74714acc2636339f53b56c4e9/meshmode/discretization/visualization.py#L771-L774
according to which `file_name` must be one of the `par_file_names`.

A better fix IMO would be to change the `par_file_names` right before inserting them into the manifest:
https://github.com/inducer/meshmode/blob/9058582b695091f74714acc2636339f53b56c4e9/meshmode/discretization/visualization.py#L771-L774
to be relative to the manifest path.

Python has [`os.path.relpath`](https://github.com/inducer/meshmode/blob/9058582b695091f74714acc2636339f53b56c4e9/meshmode/discretization/visualization.py#L771-L774) just for that purpose.

Comment (MTCam) [reply meshmode/discretization/visualization.py:644]: Did you intend to post the same exact snippet twice here?

Comment (MTCam) [reply meshmode/discretization/visualization.py]: Ok, I think this is fixed up how you wanted it now.

Comment (inducer) [reply meshmode/discretization/visualization.py]: LGTM, thanks!

**Pull request merged**